UtilityMemoryBag.lua

```diff
@@ -1,9 +1,226 @@
 -- Utility memory bag by Directsun
--- Version 2.6.0
+-- Version 2.7.0
 -- Fork of Memory Bag 2.0 by MrStump
+--
+-- Want to contribute? Create an issue or fork the code on GitHub and submit a pull request:
+-- https://github.com/sunflowermans/TTS-UtilityMemoryBag
+
+CONFIG = {
+    MEMORY_GROUP = {
+        -- This determines how many frames to wait before actually placing objects onto the table when the "Place" button is clicked.
+        -- This gives the other bags time to recall their objects.
+        -- The delay ONLY occurs if other bags have objects out.
+        FRAME_DELAY_BEFORE_PLACING_OBJECTS = 30,
+    },
+}
+
+
+--[[ Memory Bag Groups ]]-------------------------------------------------------
+--[[
+Utility Memory Bags may be added to a named group, called a "memory group".
+
+You can add a bag to a group through the bag's UI: "Setup" > "Group Name" (to the left of the bag).
+
+Only one bag from a group may have it's contents placed on the table at a time.
+When "Place" is clicked on a bag, the other bags in it's memory group are recalled.
+
+By default a memory bag is not in any group. It's memory group is "nil".
+--]]
+
+memoryGroupName = {memoryBag=self}
+function memoryGroupName:get()
+    return self._name
+end
+function memoryGroupName:set(newName)
+    GlobalMemoryGroups:unregisterBagInGroup(self:get(), self.memoryBag.getGUID())
+    GlobalMemoryGroups:registerBagInGroup(newName, self.memoryBag.getGUID())
+
+    if newName == "" then
+        self._name = nil
+    else
+        self._name = newName
+    end
+end
+
+
+-- Click the "Recall" button on all other bags in my memory group.
+function recallOtherBagsInMyGroup()
+    for bagGuid,_ in pairs(GlobalMemoryGroups:getGroup(memoryGroupName:get())) do
+        if bagGuid ~= self.getGUID() then
+            bag = getObjectFromGUID(bagGuid)
+            bag.call('buttonClick_recall')
+        end
+    end
+end
+
+-- Return "true" if another bag in my memory group has any objects out on the table.
+function anyOtherBagsInMyGroupArePlaced()
+    for bagGuid,_ in pairs(GlobalMemoryGroups:getGroup(memoryGroupName:get())) do
+        if bagGuid ~= self.getGUID() then
+            local bag = getObjectFromGUID(bagGuid)
+            local state = bag.call('areAnyOfMyObjectsPlaced')

```

Comment:
Yes it would. Great catch! I didn't consider that people might delete bags, in which case `GlobalMemoryGroups` would have stale/non-existent guids.

I'll submit a shortly.